services/QuillLMS/client/app/bundles/PremiumHub/containers/PremiumFilterableReportsContainer.tsx

```diff
@@ -0,0 +1,295 @@
+    setLastSubmittedTeachers(originalAllTeachers)
+    setLastSubmittedClassrooms(originalAllClassrooms)
+    setLastSubmittedTimeframe(defaultTimeframe(allTimeframes))
+    setLastSubmittedCustomStartDate(null)
+    setLastSubmittedCustomEndDate(null)
+    setHasAdjustedFiltersSinceLastSubmission(false)
+  }
+
+  function applyFilters() {
+    setSearchCount(searchCount + 1)
+    setLastSubmittedGrades(selectedGrades)
+    setLastSubmittedSchools(selectedSchools)
+    setLastSubmittedTeachers(selectedTeachers)
+    setLastSubmittedClassrooms(selectedClassrooms)
+    setLastSubmittedTimeframe(selectedTimeframe)
+    setLastSubmittedCustomStartDate(customStartDate)
+    setLastSubmittedCustomEndDate(customEndDate)
+    setHasAdjustedFiltersSinceLastSubmission(false)
+  }
+
+  function closeCustomDateModal() { setShowCustomDateModal(false) }
+
+  function setCustomDates(startDate, endDate) {
+    setCustomStartDate(startDate)
+    setCustomEndDate(endDate)
+    closeCustomDateModal()
+    applyFilters()
+  }
+
+  function handleClickDownloadReport() { window.print() }
+
+  if (loadingFilters) {
+    return <Spinner />
+  }
+
+  const filterProps = {
+    allTimeframes,
+    allSchools,
+    allGrades,
+    allTeachers,
+    allClassrooms,
+    applyFilters,
+    clearFilters,
+    selectedGrades,
+    setSelectedGrades,
+    hasAdjustedFiltersFromDefault,
+    handleSetSelectedTimeframe,
+    selectedTimeframe,
+    selectedSchools,
+    setSelectedSchools,
+    selectedClassrooms,
+    setSelectedClassrooms,
+    selectedTeachers,
+    setSelectedTeachers,
+    closeMobileFilterMenu,
+    showMobileFilterMenu,
+    hasAdjustedFiltersSinceLastSubmission,
+    customStartDate,
+    customEndDate,
+  }
+
+  const sharedProps = {
+    accessType,
+    loadingFilters,
+    customStartDate,
+    customEndDate,
+    pusherChannel,
+    searchCount,
+    selectedClassrooms,
+    allClassrooms,
+    selectedGrades,
+    allGrades,
+    selectedSchools,
+    selectedTeachers,
+    allTeachers,
+    selectedTimeframe,
+    handleClickDownloadReport,
+    openMobileFilterMenu
+  }
+
+  const shouldRenderDataExportContainer = location && location.pathname === '/teachers/premium_hub/data_export'
+  const shouldRenderUsageSnapshotsContainer = location && location.pathname === '/teachers/premium_hub/usage_snapshot_report'
+
+  if (accessType !== FULL) {
+    return restrictedPage
+  }
+
+  return (
+    <div className="filterable-reports-container white-background">
+      {showCustomDateModal && (
+        <CustomDateModal
+          close={closeCustomDateModal}
+          passedEndDate={customEndDate}
+          passedStartDate={customStartDate}
+          setCustomDates={setCustomDates}
+        />
+      )}
+      <Filters
+        {...filterProps}
+      />
+      {shouldRenderDataExportContainer && <DataExportContainer {...sharedProps} />}
+      {shouldRenderUsageSnapshotsContainer && <UsageSnapshotsContainer {...sharedProps} />}
```

Comment:
I think you should be able to use a `<Routes><Route>...` structure here (or a `<Switch><Route>...` if the v6 syntax doesn't work for some reason), right?